### DISCUSSION TITLE
Splitting out multiple animations from a single animation

### DISCUSSION THREAD

**Discussion by ninjz** [sceneview/sceneform-android#430] [Q&A]
I am looking to load a GLB file which contains a single animation which has multiple animations set at different times/frames.

Is there any way I can split these out and reference each animation individually? I was able to do this for my app on iOS using SceneKit however, I can't seem to find anyway to do this with this library on the ModelNode.

**Comment by AdrienBotrel** [comment]
I think you can try `modelNode.playAnimation(...)`. With this, you can play an animation depending on its rank or depending on its name

**Reply by ninjz** [reply]
Yeah, that is what I currently use because I have a glb file which has each animation named within it. I just want to align with the pipeline which I have on iOS which puts all the animations on a single timeline, and referenced by start and end frames. I'm wondering if it's possible to do that using this library

**Reply by AdrienBotrel** [reply]
I don't know if it's possible with the library.
But if you create your glb file with Blender, when you export your file as gltf, you can activate "Animation -> Animation -> Group by NLA  track" and all your animations will be grouped into one. Then you have only 1 animation to start in your app. That's how I do personnaly

**Reply by ninjz** [reply]
Yeah, the main thing is really just being able to play an animation at an arbitrary start to end frame. I also require being able to loop from those segments as well so I would want to be able to create separate player objects to control each of those.

**Reply by AdrienBotrel** [reply]
I never used that before and I don't think it's possible to do so for the moment(need confirmation from main contributors).
But it could be a good upgrade for later versions.
